debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -1244,10 +1230,80 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         // The EXACT number of instructions to return in the result.
         instruction_count: i64,
     ) -> Result<Vec<dap_types::DisassembledInstruction>, DebuggerError> {
-        let instruction_offset_as_bytes = (instruction_offset
-            * target_core.core_data.debug_info.get_instruction_size() as i64)
-            / 4
-            * 5;
+        let cs = match target_core.core.instruction_set()? {
+            InstructionSet::Thumb2 => {
+                let mut capstone_builder = Capstone::new()
+                    .arm()
+                    .mode(armArchMode::Thumb)
+                    .endian(Endian::Little);
+                if matches!(target_core.core.core_type(), CoreType::Armv8m) {
+                    capstone_builder = capstone_builder
+                        .extra_mode(std::iter::once(capstone::arch::arm::ArchExtraMode::V8));
+                }
+                capstone_builder.build()
+            }
+            InstructionSet::A32 => {
+                // TODO: We need to inspect the CPSR to determine what mode this is opearting in
+                let mut capstone_builder = Capstone::new()
+                    .arm()
+                    .mode(armArchMode::Arm)
+                    .endian(Endian::Little);
+                if matches!(target_core.core.core_type(), CoreType::Armv8m) {
+                    capstone_builder = capstone_builder.extra_mode(
+                        [
+                            capstone::arch::arm::ArchExtraMode::V8,
+                            capstone::arch::arm::ArchExtraMode::MClass,
+                        ]
+                        .into_iter(),
+                    );
+                } else if matches!(
+                    target_core.core.core_type(),
+                    CoreType::Armv6m | CoreType::Armv7m | CoreType::Armv7em
+                ) {
+                    capstone_builder = capstone_builder
+                        .extra_mode(std::iter::once(capstone::arch::arm::ArchExtraMode::MClass));
+                }
+                capstone_builder.build()
+            }
+            InstructionSet::A64 => {
+                // TODO: We need to inspect the CPSR to determine what mode this is opearting in
```

Comment:
You can remove this TODO - when I added the 32/64 bit support I solved this at another level

Reply:
And the previous one above

Reply:
Done